### ISSUE TITLE
Unsubscribe from all page: settings link doesn't work

### ISSUE THREAD

**Issue by vedmant** [ezl/dailyidea.com#961]
Just noticed that link doesn't work. As I understand it should navigate to login page if user is not authenticated?

**Comment by Watson-B** [comment]
Can you expand upon the question @vedmant? I thought you were referring to the unsubscribe from all link in emails, however that is working for me 
![image](https://user-images.githubusercontent.com/65913664/113921183-8958fb80-979a-11eb-9847-37657d61cc23.png)

![image](https://user-images.githubusercontent.com/65913664/113921141-7e9e6680-979a-11eb-9e83-44ed049ab5ee.png)

The toggle on the settings page seems to be working too


![image](https://user-images.githubusercontent.com/65913664/113921356-c1603e80-979a-11eb-8fdd-ca317468c0e0.png)

**Comment by vedmant** [comment]
@Watson-B When I click "email preferences." it's going to 404

**Comment by Watson-B** [comment]
@vedmant Ah I understand now. Since presumably the user would be coming from their email when they hit the 'unsubscribed' screen with that link, can we also treat it as a authentication and take them directly to their settings?

**Comment by vedmant** [comment]
@Watson-B We can, but that's a piece of work, let @ezl or @ikosovec create a ticket if this task should be done.

**Comment by ezl** [comment]
@vedmant the "email preferences" link in the email SHOULD authenticate the user and take them to the email preferences part of their settings.

I will be happy to create a separate ticket for this, but before we do, I think some more discussion is in order.

(A) in general, we are considering email to be a valid form of authentication. So a link from within an email should not direct a user to authenticate, as having the email is considered a sufficient proof of identity.

(B) however, we need to have a LIMIT to this -- I believe that the validity period on these tokens is NOT infinite. Can you confirm if/how long that duration of the JWT auth token validity is please @vedmant ?

**Comment by vedmant** [comment]
@ezl Sure, I see currently token expiration set to 7 days

**Comment by ezl** [comment]
> 7 days

ok, that's great.

then for the purposes of this ticket, can we just make it so that clicking the email preferences button in emails authenticates the user and takes them to the email settings of the settings page please?

if you prefer a separate ticket be created for this, lmk.

**Comment by vedmant** [comment]
@ezl Yes, I think it's out of this ticket scope, new issues makes sense.

**Comment by ezl** [comment]
@ikosovec please create a new ticket for this and close the original
